### PR TITLE
Make `compiled_data` a default feature

### PR DESCRIPTION
**Pull request by robertbastian**
#2945

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]

**Review by Manishearth — DISMISSED** [review]

**Review by Manishearth — DISMISSED** [review]

**Comment by Manishearth** [comment]
```
  error occurred: Command "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.36.32532\\bin\\HostX64\\x64\\lib.exe" "-out:D:\\a\\icu4x\\icu4x\\target\\debug\\build\\harfbuzz-sys-4698af6d23557bbd\\out\\libembedded_harfbuzz.a" "-nologo" "D:\\a\\icu4x\\icu4x\\target\\debug\\build\\harfbuzz-sys-4698af6d23557bbd\\out\\harfbuzz/src/harfbuzz.o" with args "lib.exe" did not execute successfully (status code exit code: 1108).
```

**Comment by Manishearth** [comment]
It fails in my PR too, fwiw https://github.com/unicode-org/icu4x/actions/runs/5445404819/jobs/9904622941?pr=3616

File: ffi/gn/Cargo.lock [hunk 1]
```diff
@@ -214,17 +214,21 @@ name = "icu"
 version = "1.2.0"
 dependencies = [
  "icu_calendar",
+ "icu_casemapping",
```
[the remark this answers is not in the document]
Comment (robertbastian) [reply ffi/gn/Cargo.lock]: fixed

File: components/datetime/Cargo.toml [hunk 2]
```diff
@@ -48,7 +48,7 @@ serde = { version = "1.0", default-features = false, features = ["derive", "allo
 databake = { version = "0.1.3", path = "../../utils/databake", features = ["derive"], optional = true}
```
[the remark this answers is not in the document]
Comment (robertbastian) [reply components/datetime/Cargo.toml:49]: fixed

**Review by Manishearth — APPROVED** [review]

**Review by sffc — APPROVED** [review]
Thanks for adding the deps test. The rest looks like what I expect to see.

**Review by Manishearth — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**